webrender/build.rs

```diff
@@ -216,9 +223,47 @@ fn create_shaders(glsl_files: Vec<PathBuf>, out_dir: String) {
             } else {
                 file_name.push_str(".frag");
             }
-            let file_path = Path::new(&out_dir).join(file_name);
-            let mut file = File::create(file_path).unwrap();
+            let file_path = Path::new(&out_dir).join(&file_name);
+            let mut file = File::create(&file_path).unwrap();
             write!(file, "{}", shader_source).unwrap();
+            file_name_vector.push(file_name);
+        }
+    }
+    return file_name_vector;
+}
+
+#[cfg(any(target_os = "windows"))]
+fn compile_fx_files(file_name_vector: Vec<String>, out_dir: String) {
+    for mut file_name in file_name_vector {
+        let is_vert = file_name.ends_with(".vert");
+        let supported = SUPPORTED_SHADERS.iter().any(|s| file_name.contains(s));
+        let file_path = Path::new(&out_dir).join(&file_name);
+        if cfg!(target_os = "windows") && supported {
+            file_name.push_str(".fx");
+            let fx_file_path = Path::new(&out_dir).join(&file_name);
+            //let sdk_path = env::var("DXSDK_DIR").ok().expect("Please set the DXSDK_DIR enviroment variable");
+            //let sdk_path = PaŁth::new(&sdk_path);
+            let pf_path = env::var("ProgramFiles(x86)").ok().expect("Please set the ProgramFiles(x86) enviroment variable");
```

Comment:
I wonder if we can ship `fxc.exe` with WR.
Also, might be worth checking if the program files path can be extracted from Windows registry